### DISCUSSION TITLE
Get client remote address when singup.

### DISCUSSION THREAD

**Discussion by phongjimmy** [supabase/supabase#10842] [Questions]
In my application flow, I create a member table and trigger function after inserting from auth.users table.
so I need to store the client's remote address using PostgreSQL support in currently using inet_client_addr() but the IP address is not correct (always 127.0.0.1/32)

I saw in Auth log request signup with parameters like the below JSON
```{
  "auth_event": {
    "action": "login",
    "actor_id": "77b72bf3-c73f-4158-8e7e-190e589d7cda",
    "actor_username": "aaaa@gmail.com",
    "log_type": "account",
    "traits": {
      "provider": "email"
    }
  },
  "component": "api",
  "duration": 98687530,
  "level": "info",
  "method": "POST",
  "msg": "request completed",
  "path": "/signup",
  "referer": "http://localhost:3000/",
  "remote_addr": "xxx.xx.xxx.xx",
  "status": 200,
  "time": "2022-12-10T20:42:03Z",
  "timestamp": "2022-12-10T20:42:03Z"
}```

how to get remote_addr and store it in the member table after the user registered

**Comment by GaryAustin1** [comment]
I don't believe you can on the signup call.
But you can follow up with a database request thru the API and PostgREST gives you access to all the headers and could get it from that.
Of course there is a chance the user finishes signing in on a different device location if you use email confirmation.

https://postgrest.org/en/stable/api.html#http-context shows how to get the headers.

**Reply by phongjimmy** [reply]
Hello, @GaryAustin1  thanks for the reply.
can you make an example code for this? or any tutorial

**Reply by GaryAustin1** [reply]
I've never seen anyone do this.  Just saying if you can make a database call you can get access to the headers and return them.
You can see the available headers in your API_EDGE_NETWORK_LOG

So you could put it in an rpc function that returns text.
And do something like:
` select current_setting('request.headers', true)::json->>'x-real-ip'::text`
![image](https://user-images.githubusercontent.com/54564956/206960971-a6665072-59a7-4d03-aa6f-84b3ffb00a83.png)

And you use `const {data, error} = await supabase.rpc('get_header')` to call it.

**Reply by phongjimmy** [reply]
Does it look like works from the client side but not in the PostgreSQL function.

My code is
```
create table public.members (
  id            uuid references auth.users on delete cascade not null primary key,
  username      varchar(128) unique,
  firstname     varchar(255),
  lastname      varchar(255),
  avatar        varchar(255),
  ip_address    varchar(32),
  status        memeber_status_type not null default 'OFFLINE'::memeber_status_type,
  updated_at    timestamp with time zone,
  inserted_at   timestamp with time zone default timezone('utc'::text, now()) not null
);
```

this table is additional for the user, and function to trigger

```
create function handle_new_user_member()
  returns trigger as
  $$
    declare
      clientip varchar(32);
    begin
      
    ....

      select current_setting('request.headers', true)::json->>'x-real-ip'::text into clientip;
      if (clientip is not null) then
        update public.members set ip_address = clientip where id = new.id;
      end if;

      ....
      return new;
    end
  $$
language plpgsql security definer;

drop trigger if exists on_auth_user_created_profile on auth.users;
create trigger on_auth_user_created_profile
  after insert on auth.users
  for each row execute procedure public.handle_new_user_member();
```


what I wanna is to store client's IP address in the member table using PostgreSQL function normally  without the client side
in a current way I embed the IP address in raw_user_meta_data, but what I mean is to get the client IP in PostgreSQL
my project doest not using any backend, only reactjs + supabase

**Reply by GaryAustin1** [reply]
No, as I said you would have to use an rpc call from the client.  This only works with a PostgREST database call, not auth.    It does not have to pass any value and you can update your user table in that rpc call, so the user can’t fake it.
